steps/delius/upw/create-upw-project.ts

```diff
@@ -0,0 +1,92 @@
+            startTime: '09:00',
+            endTime: '17:00',
+        },
+    }: {
+        providerName: string
+        teamName: string
+        projectType?: string
+        pickupPoint?: string
+        projectName?: string
+        projectCode?: string
+        projectAvailability?: {
+            day: string
+            frequency: string
+            startDate: Date
+            endDate: Date
+            startTime: string
+            endTime: string
+        }
+    }
+): Promise<{ projectCode: string; projectName: string }> {
+    await page.getByRole('link', { name: 'UPW Projects' }).click()
+    await expect(page.locator('#content > h1')).toContainText('UPW Projects List')
+    await page.locator('input', { hasText: 'Add New Project' }).click()
+
+    await expect(page.locator('#content > h1')).toContainText('Add Project')
+    await selectOption(page, '#Trust\\:selectOneMenu', providerName)
+    await selectOption(page, '#Team\\:selectOneMenu', teamName)
+    await page.selectOption('#ProjectType\\:selectOneMenu', projectType)
+    await page.selectOption('#DefaultPickupPoint\\:selectOneMenu', { label: pickupPoint })
+    await page.fill('#ProjectCode\\:prependedInputText', projectCode)
+    await page.fill('#ProjectName\\:inputText', projectName)
+    await fillDate(page, '#ProjectStartDate\\:datePicker', new Date())
+    await page.getByRole('button', { name: 'Save' }).click()
+    await expect(page.locator('#content > h1')).toContainText('Update Project')
+    await addProjectAvailability(page, projectAvailability)
+    return { projectCode, projectName }
+}
+
+export function createNameWithTimeStamp(prefix = 'project'): string {
+    const now = new Date()
+    const dateFormat =
+        now.getDate().toString().padStart(2, '0') +
+        (now.getMonth() + 1).toString().padStart(2, '0') +
+        now.getFullYear() +
+        now.getHours().toString().padStart(2, '0') +
+        now.getMinutes().toString().padStart(2, '0') +
+        now.getSeconds().toString().padStart(2, '0')
+    return `${prefix}-${dateFormat}`
+}
+
+async function addProjectAvailability(
+    page: Page,
+    projectAvailability: {
+        day: string
+        frequency: string
+        startDate: Date
+        endDate: Date
+        startTime: string
+        endTime: string
+    }
+) {
+    await page.getByRole('button', { name: 'Project Availability' }).click()
+    await selectOption(page, '#Day\\:selectOneMenu', projectAvailability.day)
+    await selectOption(page, '#Frequency\\:selectOneMenu', projectAvailability.frequency)
+    await fillDate(page, '#StartDate\\:datePicker', projectAvailability.startDate)
+    await fillDate(page, '#EndDate\\:datePicker', projectAvailability.endDate)
+    await page.fill('#StartTime\\:timePicker', projectAvailability.startTime)
+    await page.fill('#EndTime\\:timePicker', projectAvailability.endTime)
+    await page.locator('input[type="submit"][name="j_idt808"][value="Add"]').click()
```

Comment:
Those `j_id*` IDs and names are auto-generated and can change between releases, you're probably better off using something like:
```suggestion
    await page.getByRole('button', { name: 'Add' }).click()
```

Playwright recommends testing user-visible properties, rather than complicated selectors: https://playwright.dev/docs/best-practices#use-locators.  I find it's handy to just use the Playwright recorder, as it generates these for you.

Reply:
Actually the reason I had to do this was due to there being 2 other "Add" buttons in there - let me see If I have a work around, so I don't use the autogenerated name

Reply:
Ah, sorry - hadn't realised that.  This is what the Playwright recorder suggests:

```typescript
    await page.getByTitle('Add Availability').click()
```

Reply:
Thanks, that worked!